src/screen/GameScreen.java

```diff
@@ -790,8 +798,15 @@ && checkCollision(bullet, this.enemyShipSpecial)) {
 				if(item.getSpriteType() == SpriteType.PerpleEnhanceStone){
 					this.enhanceManager.PlusNumEnhanceStoneAttack(1);
 				}
-				this.ship.checkGetItem(item);
+				/* version 2.0 : when item is destroyed, show effect */
+				if (!item.isDestroyed()) {
+					this.ship.checkGetItem(item);
+					item.destructionCooldown.reset();
+					this.logger.info("Get Item ");
+				}
 			}
+			if (item.isDestroyed() && item.destructionCooldown.checkFinished())
```

Comment:
[P3] Smart way of showing item get effect for a certain amount of time using Cooldown